doc/developer/platform/architecture-storage.md

```diff
@@ -107,25 +107,142 @@ All commands are durably persisted and their effect survives crashes and reboots
 
 ## Source ingestion
 
-In the case where the controller is requested to ingest a source into a
-collection, an ingestion pipeline must be constructed. When running in the
-cloud setting each ingestion instance for a particular source will be hosted by
-a dedicated kubernetes pod but when running in the single binary `materialized`
-setting each ingestion instance will translate to a particular timely dataflow
-being deployed to the STORAGE timely cluster.
-
-For external users of STORAGE, the only thing they will be able to see is the
-definite collection containing the result of the ingested data after all
-requested processing has been performed (e.g decoding and envelope processing).
-
-However, there are good reasons for STORAGE to internally create and manage
-multiple colletion that work together to form the final collection. For
-example, in the case where STORAGE is instructed to ingest a source that
-requires Avro decoding and DEBEZIUM envelope processing, STORAGE might decide
-to first create a definite collection of the raw data before decoding and
-envelope processing and compute the final collection based on that. This
-ensures that STORAGE will always have the ability to fix bugs in the Avro
-decoding logic and generate new renditions based on the original raw data that
-have been safely stored in the raw collection.
+In response to a CREATE SOURCE command, the STORAGE layer should instantiate a *source ingestion pipeline*.
+This pipeline does not need to be a dataflow, but as we will see it is helpful to use timestamps to maintain consistency.
+
+The goal of the source ingestion pipeline is to produce data that can be appended to a `persist` collection.
```

Comment:
it's unclear to me whether you mean a persist _shard_ or a storage _collection_ here (and elsewhere). sorry for the nitpick, but I think it's especially important that we use consistent language in our architecture docs to avoid confusion and those are the terms used in the storage and persist design docs

my guess from context is that this one is actually a storage collection, since below about it being the "output of the source ingestion pipeline" and that might be sharded (e.g. by kafka partition) and have renditions (e.g. if we change decoding/envelope logic)

Reply:
That's a good point. As long as "storage collection" is understood to mean "persistent collection" I can just use that!